From: SeongJae Park <sj@kernel.org>
Date: Mon, 20 Jan 2025 13:43:38 -0800
Subject: [PATCH] ==== auto-tune monitoring parameters ====

Aggregation interval should be appropriate.  If it is too short,
everything looks cold.  It's especially bad because ARA works with
nr_accesses only.  If it is too long, everything looks hot.

Auto-tune intervals aiming appropriate amount of hotnesss we get.  Use
two metrics.

1. Positive access check samples ratio.  20% by default.  If there are
   100 regions and max_nr_accesses is 20, total number of samples for
   the aggregation interval is 2000.  We aim to have at least 400
   positive check results.
2. Total size of regions having >0 nr_accesses.  20% by default.

Aim to meet both condition.  Prefer longer aggregation interval.  So the
unified metric is min(metric_a, metric_b).  Reuse the proportional
feedback loop.  Keep max_nr_accesses fixed.  That is, when increase
aggregation interval, increase sampling interval in same proportion.

Next question: what about monitoring overhead?

Let user sets the max CPU usage.  Calculate average overhead of single
access check and adjust max_nr_regions to meet the limit.

Should we also set the min CPU usage?  No, accuracy is defined by
max_nr_accesses and the two auto-tuning aim metrics.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_448v7nrp | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_448v7nrp

diff --git a/hkml_cv_bogus/hkml_cv_bogus_448v7nrp b/hkml_cv_bogus/hkml_cv_bogus_448v7nrp
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

